worlds/spire/__init__.py

```diff
@@ -63,15 +64,13 @@ def create_regions(self):
 
     def fill_slot_data(self) -> dict:
         slot_data = {
-            'seed': "".join(self.multiworld.per_slot_randoms[self.player].choice(string.ascii_letters) for i in range(16))
+            'seed': "".join(self.random.choice(string.ascii_letters) for i in range(16))
         }
-        for option_name in spire_options:
-            option = getattr(self.multiworld, option_name)[self.player]
-            slot_data[option_name] = option.value
+        slot_data.update(self.options.as_dict("character", "ascension", "final_act", "downfall"))
```

Comment:
Maybe it's the desired behavior, so feel free to dismiss my comment.

 If I was maintaining this world, I would be slightly annoyed that option names need to be repeated here to be added to `slot_data`, since it's in a different file than the option definition.

Reply:
With the dataclass, it would be adding a lot of other options that weren't being added before if the old method was used, this explicitly adds only the options that were previously included